gef-ui/server/controllers/bank-review-date/get-bank-review-date.ts

```diff
@@ -0,0 +1,92 @@
+import { Response } from 'express';
+import { getDate, getMonth, getYear, parseISO } from 'date-fns';
+import { CustomExpressRequest, isFacilityEndDateEnabledOnGefVersion, parseDealVersion } from '@ukef/dtfs2-common';
+import * as api from '../../services/api';
+import { BankReviewDateViewModel } from '../../types/view-models/bank-review-date-view-model';
+import { asLoggedInUserSession } from '../../utils/express-session';
+import { Facility } from '../../types/facility';
+
+type BankReviewDateParams = { dealId: string; facilityId: string };
+
+type GetBankReviewDateRequest = CustomExpressRequest<{ params: BankReviewDateParams; query: { status: string | undefined } }>;
+
+type HandleGetBankReviewDateParams = {
+  req: GetBankReviewDateRequest;
+  res: Response;
+  previousPage: string;
+};
+
+/**
+ * @param facility - the facility
+ * @param previousPage - the previous page url
+ * @param status - the query parameter status
+ * @returns view model for the bank review date template
+ */
+const getBankReviewDateViewModel = (facility: Facility, previousPage: string, status: string | undefined): BankReviewDateViewModel => {
+  if (typeof facility.dealId !== 'string' || typeof facility._id !== 'string') {
+    throw new Error('Invalid facility or deal id provided');
+  }
```

Comment:
Oops, this was added before I added the `Facility` type. It all boils down to whether we trust the api response!

Reply:
It seems simpler to remove this I think @MarRobSoftwire but happy to keep in if you think this might be a scenario that happens. 

I'm not sure how or if the API response would ever return a deal or facility ID that is _not_ a string?

Reply:
Removed